concent_api/api-integration-additional-verification-test.py

```diff
@@ -0,0 +1,381 @@
+#!/usr/bin/env python3
+
+import hashlib
+import os
+import sys
+import time
+from freezegun import freeze_time
+
+from golem_messages import message
+
+from utils.helpers import get_current_utc_timestamp
+from api_testing_common import api_request
+from api_testing_common import assert_condition
+from api_testing_common import count_fails
+from api_testing_common import create_client_auth_message
+from api_testing_common import create_signed_task_to_compute
+from api_testing_common import get_task_id_and_subtask_id
+from api_testing_common import PROVIDER_PRIVATE_KEY
+from api_testing_common import PROVIDER_PUBLIC_KEY
+from api_testing_common import REQUESTOR_PRIVATE_KEY
+from api_testing_common import REQUESTOR_PUBLIC_KEY
+from api_testing_common import run_tests
+from api_testing_common import timestamp_to_isoformat
+from api_testing_common import upload_file_to_storage_cluster
+
+import requests
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+
+def get_subtask_results_verify(
+    task_id,
+    subtask_id,
+    current_time,
+    cluster_consts,
+    reason,
+    report_computed_task_size,
+    report_computed_task_package_hash,
+    task_to_compute_size,
+    task_to_compute_package_hash,
+):
+    task_to_compute = create_signed_task_to_compute(
+        task_id=task_id,
+        subtask_id=subtask_id,
+        deadline=current_time + cluster_consts.additional_verification_call_time,
+        price=1,
+        size=task_to_compute_size,
+        package_hash=task_to_compute_package_hash,
+    )
+
+    report_computed_task = message.ReportComputedTask(
+        task_to_compute=task_to_compute,
+        subtask_id=subtask_id,
+        size=report_computed_task_size,
+        package_hash=report_computed_task_package_hash,
+    )
+
+    with freeze_time(timestamp_to_isoformat(current_time)):
+        subtask_results_rejected = message.tasks.SubtaskResultsRejected(
+            reason=reason,
+            report_computed_task=report_computed_task,
+        )
+        subtask_results_rejected.sign_message(
+            REQUESTOR_PRIVATE_KEY,
+            subtask_results_rejected.get_short_hash(),
+        )
+
+        subtask_results_verify = message.concents.SubtaskResultsVerify(
+            subtask_results_rejected=subtask_results_rejected,
+        )
+
+    return subtask_results_verify
+
+
+@count_fails
+def test_case_1_test_for_positive_case(cluster_consts, cluster_url, test_id):
```

Comment:
In all tests I see that `SubtaskResultsVerify` is being send from the requstor instead of provider. Please change / clarify.

Reply:
@dybi I have checked now and it looks like you are right, it should be provider. However, I have changed this in the use case implementation too in https://github.com/golemfactory/concent/pull/436, it went throught review and was approved. So this can be resolved in on of two ways:
- You can either revert the commits that change provider to requestor, or if those commits also contain other changes, do it manually in new PR. All the changes are in the mentioned PR so it should not take much time.
- Wait until Monday for me to do it.

Reply:
@rwrzesien - Already solved and created new PR :)

Reply:
:+1: